src/liballoc/rc.rs

```diff
@@ -571,9 +571,14 @@ impl<T: ?Sized> Rc<T> {
     /// ```
     #[stable(feature = "rc_raw", since = "1.17.0")]
     pub fn into_raw(this: Self) -> *const T {
-        let ptr: *const T = &*this;
+        let ptr: *mut RcBox<T> = NonNull::as_ptr(this.ptr);
+        let fake_ptr = ptr as *mut T;
         mem::forget(this);
-        ptr
+
+        unsafe {
```

Comment:
I think there's useful commentary from your PR description that could be added here as well as some `// SAFETY` notes. (Similarly for the other file.)

Reply:
I have added said notes.